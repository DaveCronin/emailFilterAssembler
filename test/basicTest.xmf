#Verify allowed variable names
a = {to "test"}
abcdefghijklmnopqrstuvwxyz = {to "test"}
1 = [read]
1234567890 = [read]

# Verify each Matcher attribute gets set correctly
testFrom = {from "testFrom"}
testTo = {to "testTo"}
testSub = {subject "testSub"}
testCon = {contains "testCon"}
testNotCon = {notContains "testNotCon"}

# Verify each Action attribute gets set correctly
testRead1 = [read]
testRead2 = [mark read]
testRead3 = [mark as read]
testStar = [star]
testDel = [delete]
testArchive = [archive]
testSpam1 = [not spam]
testSpam2 = [never spam]
testImp1 = [important]
testImp2 = [mark important]
testImp3 = [mark as important]


if a -> 1
if abcdefghijklmnopqrstuvwxyz -> 1234567890
if testFrom -> 1
if testTo -> 1
if testSub -> 1
if testCon -> 1
if testNotCon -> 1
if a -> testRead1
if a -> testRead2
if a -> testRead3
if a -> testStar
if a -> testDel
if a -> testArchive
if a -> testSpam1
if a -> testSpam2
if a -> testImp1
if a -> testImp2
if a -> testImp3
