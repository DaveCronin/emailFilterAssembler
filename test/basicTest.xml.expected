<?xml version='1.0' encoding='UTF-8'?><feed xmlns='http://www.w3.org/2005/Atom' xmlns:apps='http://schemas.google.com/apps/2006'>
	<title>Mail Filters</title>
	<id>tag:mail.google.com,2008:filters:4366218717592519022</id>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='from' value='testFrom'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='testTo'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='subject' value='testSub'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='hasTheWord' value='testCon'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='doesNotHaveTheWord' value='testNotCon'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldMarkAsRead' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldStar' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldTrash' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldArchive' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldNeverSpam' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldNeverSpam' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldAlwaysMarkAsImportant' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldAlwaysMarkAsImportant' value='true'/>
	</entry>
	<entry>
		<category term='filter'></category>
		<title>Mail Filter</title>
		<id>tag:mail.google.com,2008:filter:4366218717592519022</id>
		<apps:property name='to' value='test'/>
		<apps:property name='shouldAlwaysMarkAsImportant' value='true'/>
	</entry>
</feed>